### DISCUSSION TITLE
Non-descript OIDC error

### DISCUSSION THREAD

**Discussion by wreality** [outline/outline#7663] [Self Hosting] [answered]
I'm trying to diagnose an authentication error with Authentik and Outline.  I have one instance that's working fine, but another Outline/Authentik pair that goes returns this after redirecting and returning from Authentik:

```
{"label":"http","level":"info","message":"  <-- GET /auth/oidc.callback?code=a56d5b25a342404a9214702378651059&state=3cfc951beb052be1"}
{"level":"error","message":"Error during authentication"}
{"label":"http","level":"info","message":"  --> GET /auth/oidc.callback?code=a56d5b25a342404a9214702378651059&state=3cfc951beb052be1 302 142ms 69b"}
{"label":"http","level":"info","message":"  <-- GET /?notice=auth-error"}
{"label":"http","level":"info","message":"  --> GET /?notice=auth-error 200 12ms 1.47kb"}                                                                             
```

I've tried enabling debug, increasing verbosity, even attaching to node with the debugger to try and understand where I've missed something / get more context to debug what's going on.  

If anyone's experienced this, or has any other pointers, I'd be most appreciative.

**Comment by tommoor** [comment]
Did you try `LOG_LEVEL=debug` specifically? The full error should be visible here

**Reply by wreality** [reply, marked as answer]
This may seem incredulous, but the log output above is with `LOG_LEVEL=debug` set. :-)

This did spur me into another round of troubleshooting.  The end issue was missing `/` at the end of the token URL.  This may be (read: is likely) authentik specific, but when the URL is missing the slash Authentik responds to the token request with a redirect, which Passport/Node follows, but when it follows, that results in the request being a GET and not POST, which Authentik returns with a 405.

I was able to find the error with some node.js debugging:
![image](https://github.com/user-attachments/assets/464cf0d8-bd5a-46bf-90dd-4b0dc5e90fd3)

So, no fault of Outline.  The useful parts of the error message *did* get swallowed, so there may be some improvements to be had there.

The error swallowing happens at the link below, but fixing it to display the error may not be of huge benefit/would probably result in some gymnastics.
 
https://github.com/outline/outline/blob/ffae5d2f209963bf08ca18ee561c1f11b866cad3/server/middlewares/passport.ts#L23
